_drafts/wework-com-is-going-static.markdown

```diff
@@ -0,0 +1,113 @@
+---
+layout:       post
+title:        WeWork.com is Going Static
+author:       ramin_bozorgzadeh
+summary:      Back when the web first started, things were a lot simpler. Most websites were made up of static html pages and there weren't a lot of moving parts. This is the story of how wework.com went from a full-stack "web app" to a statically generated site, and why ...
+image:        //res.cloudinary.com/wework/image/upload/c_fill,f_auto,g_faces,h_800,w_1000/v1448734984/engineering/wework-com-is-going-static.jpg
+categories:   engineering
+---
+
+### What is this "static" business all about?
+
+Static site generators have become a hot topic in the past few years. There are [whole sites](https://www.staticgen.com/) dedicated to tracking and rating them. Don't take my word
+for it, check out this graph from [Google Trends for "static site generator"][2]:
+
+[![Google Trends - Static Site Generator][1]][2]
+[1]: //res.cloudinary.com/wework/image/upload/f_auto/v1448735896/engineering/static-site-generator.png
+[2]: https://www.google.com/trends/explore#q=static%20site%20generator
+
+The idea is actually quite simple. Whereas on a traditional *dynamic* site, each page is proceed and generated on the server per request (let's ignore caching strategies for the sake of argument), a static site is processed and generated just **once** on each deploy and all the server has to do is serve up the resulting generated HTML. Of course, this is an overly simplified explantion of how it all works, and this process doesn't work on all web sites, but it does work really well for marketing and informational sites like wework.com. This very blog you are reading is actually a statically generated site using [Jekyll](http://jekyllrb.com/) and hosted on [Github pages](https://pages.github.com/)!
+
+If you consider the amount of information that changes on a daily, or even weekly basis on a site like wework.com, it is actually quite wasteful to have a server process each and every request that comes through. One of the main reasons we even have a server-side component for wework.com is to allow us to easily add, update and manage information on the site. We can still achieve this with a statically generated site by utilizing an API to serve up the data and take the load off of our application server for serving up the html pages themselves.
```

Comment:
hmmm. maybe, although I don't want to link to them just yet cause we haven't started using them. it would be a lie! we may still consider Netlify's open source CMS: https://github.com/netlify/netlify-cms